bundles/org.openhab.binding.plex/src/main/java/org/openhab/binding/plex/internal/handler/PlexPlayerHandler.java

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.plex.internal.handler;
+
+import static org.openhab.binding.plex.internal.PlexBindingConstants.*;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.plex.internal.PlexBindingConstants;
+import org.openhab.binding.plex.internal.config.PlexPlayerConfiguration;
+import org.openhab.binding.plex.internal.dto.MediaContainer.MediaType;
+import org.openhab.binding.plex.internal.dto.PlexPlayerState;
+import org.openhab.binding.plex.internal.dto.PlexSession;
+import org.openhab.core.library.types.PlayPauseType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.*;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PlexBindingConstants} class defines common constants, which are
+ * used across the whole binding.
+ *
+ * @author Brian Homeyer - Initial contribution
+ */
+@NonNullByDefault
+public class PlexPlayerHandler extends BaseThingHandler {
+    private @NonNullByDefault({}) String playerID;
+
+    private @Nullable PlexServerHandler bridgeHandler;
+    private PlexSession currentSessionData;
+    private boolean foundInSession;
+
+    private final Logger logger = LoggerFactory.getLogger(PlexPlayerHandler.class);
+
+    private PlexApiConnector plexAPIConnector;
+
+    public PlexPlayerHandler(Thing thing) {
+        super(thing);
+        currentSessionData = new PlexSession();
+        plexAPIConnector = new PlexApiConnector(scheduler);
+    }
+
+    /**
+     * Initialize the player thing, check the bridge status and hang out waiting
+     * for the session data to get polled.
+     */
+    @Override
+    public void initialize() {
+        PlexPlayerConfiguration config = getConfigAs(PlexPlayerConfiguration.class);
+        foundInSession = false;
+        playerID = config.playerID;
+        Bridge bridge = getBridge();
+        logger.debug("Initializing PLEX player : {}", playerID);
+        ThingStatus bridgeStatus = bridge != null ? bridge.getStatus() : null;
+        boolean bridgeOnline = bridgeStatus == ThingStatus.ONLINE;
+        if (bridgeOnline) {
+            bridgeHandler = (PlexServerHandler) getBridge().getHandler();
+            if (bridgeHandler != null) {
+                updateStatus(ThingStatus.UNKNOWN);
+            }
+        } else {
+            updateStatus(ThingStatus.OFFLINE);
+        }
```

Comment:
This is actually handled by the framework. You need to set the thing status to UNKNOWN in this method. Then, the Thing is automatically set ONLINE/OFFLINE according the bridge's status.